scripts/ci/VM-container-tests.sh

```diff
@@ -0,0 +1,249 @@
+
+	if [[ $FORCE == true ]]
+	then
+		bitbake -c clean multiconfig:container:trustx-core
+		bitbake -c clean cmld
+		bitbake -c clean trustx-cml-initramfs
+		bitbake -c clean trustx-cml
+	fi
+  if [[ $BRANCH != "" ]]
+  then
+	  # TODO \${BRANCH} is defined in init_ws.sh -> if changes there, this won't work
+	  sed -i "s/branch=\${BRANCH}/branch=$BRANCH/g" cmld_git.bbappend
+	fi
+  bitbake multiconfig:container:trustx-core
+	bitbake trustx-cml
+else
+	if [ ! -d "out-yocto" ]
+	then
+		echo "ERROR: Project setup not complete. Try with --compile?"
+		exit
+	fi
+	cd out-yocto
+fi
+
+# Check if cmld was build
+if [ -z $(ls -d tmp/work/core**/cmld/git**/git) ]
+then
+  echo "ERROR: No cmld build found: did you compile?"
+  exit
+fi
+
+# Check if the branch matches the built one
+if [[ $BRANCH != "" ]]
+then
+  BUILD_BRANCH=$(git -C tmp/work/core*/cmld/git*/git branch | grep '*' | awk '{ print $NF }')  # check if git repo found and correct branch used
+  if [[ $BRANCH != $BUILD_BRANCH ]]
+  then
+    echo "ERROR: The specified branch \"$BRANCH\" does not match the build ($BUILD_BRANCH). Please recompile with flag -c."
+    exit
+  fi
+fi
+
+# Create image
+# -----------------------------------------------
+echo "STATUS: Creating images"
+rm -f containers.btrfs
+dd if=/dev/zero of=containers.btrfs bs=1M count=10000 &> /dev/null
+mkfs.btrfs -L containers containers.btrfs &> /dev/null
+
+# Backup system image
+# TODO it could have been modified if VM run outside of this script with different args already
+cp tmp/deploy/images/genericx86-64/trustme_image/trustmeimage.img trustmeimage.img
+
+# Start VM
+# -----------------------------------------------
+echo "STATUS: Starting VM"
+PROCESS_NAME="qemu-trustme-t"
+if [[ $(pgrep $PROCESS_NAME) != "" ]]
+then
+  echo "WARNING: VM instance called \"$PROCESS_NAME\" already running. Please stop/kill it first."
+  exit
+fi
+
+# copy for faster startup
+cp /usr/share/OVMF/OVMF_VARS.fd .
+
+qemu-system-x86_64 -machine accel=kvm,vmport=off -m 1024G -smp 4 -cpu host -bios OVMF.fd \
+  -name trustme-tester,process=qemu-trustme-t \
+  -nodefaults -nographic \
+	-device virtio-rng-pci,rng=id -object rng-random,id=id,filename=/dev/urandom \
+	-device virtio-scsi-pci,id=scsi -device scsi-hd,drive=hd0 \
+	-drive if=none,id=hd0,file=trustmeimage.img,format=raw \
+	-device scsi-hd,drive=hd1 \
+ 	-drive if=none,id=hd1,file=containers.btrfs,format=raw \
+	-device e1000,netdev=net0 -netdev user,id=net0,hostfwd=tcp::$SSH_PORT-:22 \
+	-drive "if=pflash,format=raw,readonly=on,file=/usr/share/OVMF/OVMF_CODE.fd" \
+	-drive "if=pflash,format=raw,file=./OVMF_VARS.fd" \
+  $VNC \
+  $TELNET &
+  # -serial mon:stdio -display curses
+
+ssh-keygen -f "/home/pecholt/.ssh/known_hosts" -R "[localhost]:$SSH_PORT" &> /dev/null
+
+# Waiting for VM's ssh server to start and get cert
+sleep 1
+echo "STATUS: Waiting for VM ssh server to start up..."
+ssh-keyscan -T 70 -p $SSH_PORT -H localhost  >> ~/.ssh/known_hosts
```

Comment:
use -o "StrictHostKeyChecking no" in ssh command instead of modyfing the user's known_hosts file